### PR TITLE
[lldb] Highlight "note:" in CommandReturnObject

### PR DESCRIPTION
**Pull request by JDevlieghere**
We have helpers to emit warnings and errors. Do the same thing for notes to they stand out more.

### CODE REVIEW

**Comment by JDevlieghere** [comment]
Motivated by #114608

File: lldb/include/lldb/Interpreter/CommandReturnObject.h [hunk 1]
```diff
@@ -110,6 +110,11 @@ class CommandReturnObject {
   void AppendMessageWithFormat(const char *format, ...)
       __attribute__((format(printf, 2, 3)));
 
+  void AppendNote(llvm::StringRef in_string);
+
+  void AppendNoteWithFormat(const char *format, ...)
+      __attribute__((format(printf, 2, 3)));
```
Comment (augusto2112) [inline lldb/include/lldb/Interpreter/CommandReturnObject.h:116]: Nice, didn't know about `__attribute__((format(printf`

File: lldb/source/Interpreter/CommandReturnObject.cpp [hunk 2]
```diff
@@ -27,6 +27,12 @@ static llvm::raw_ostream &warning(Stream &strm) {
          << "warning: ";
 }
 
+static llvm::raw_ostream &note(Stream &strm) {
```
Comment (augusto2112) [inline lldb/source/Interpreter/CommandReturnObject.cpp:30]: Since `note` is a function should it be capitalized to follow the LLDB style?

Comment (augusto2112) [reply lldb/source/Interpreter/CommandReturnObject.cpp:30]: Actually `error` and `warning` are not capitalized, I guess we can either keep all three lower case or capitalize all of them, up to you

**Review by augusto2112 — APPROVED** [review]

**Review by kastiglione — APPROVED** [review]

**Review by adrian-prantl — APPROVED** [review]
I was going to say this is not the right way to do this, but I suppose there isn't really a better mechanism available, since this isn't an error. Otherwise you'd be better off joining a diagnostic error with the note so the UI layer can choose how to render the note.

**Pull request merged**